applications/nrf_desktop/doc/battery_charger.rst

```diff
@@ -0,0 +1,53 @@
+.. _battery_charger:
+
+Battery charger module
+######################
+
+The ``battery_charger`` module is responsible for battery charging control.
+
+Module Events
+*************
+
++------------------+---------------------+---------------------+-------------------------+------------------+
+| Source Module    | Input Event         | This Module         | Output Event            | Sink Module      |
++==================+=====================+=====================+=========================+==================+
+|                  |                     | ``battery_charger`` | ``battery_state_event`` | :ref:`led_state` |
++------------------+---------------------+                     +-------------------------+------------------+
+| :ref:`usb_state` | ``usb_state_event`` |                     |                         |                  |
++------------------+---------------------+---------------------+-------------------------+------------------+
+
+Configuration
+*************
+
+The module implemented in ``battery_charger.c`` uses the Zephyr :ref:`zephyr:gpio` driver to control and monitor battery charging. For this reason, you should set :option:`CONFIG_GPIO` option.
+
+Set option ``CONFIG_DESKTOP_BATTERY_CHARGER_DISCRETE`` to enable the module (by default ``CONFIG_DESKTOP_BATTERY_CHARGER_NONE`` is selected - module is disabled).
+
+Following module configuration options are available:
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PIN`` - pin to which CSO (charging status) output from charger is connected
+* CSO pin pull:
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_NONE`` - disabled (default option)
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_UP`` - up
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_DOWN`` - down
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_PIN`` - pin which enables battery charging
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED`` - inverse charging enable signal
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_FREQ`` - CSO pin state switching fequency when charging error occurrs [Hz]
+
+Implementation details
+**********************
+
+The module disables battery charging when USB is suspended (required by USB specification) or disconnected.
+Charging is enabled when USB is powered or active.
+
+.. warning:
+  Hardware configuration should enable charging by default (e.g. when pin to enable charging is not configured).
+  Otherwise it may be impossible to enable charging when battery is completely discharged.
+
+The module checks battery state using :c:type:`struct k_delayed_work`.
+The check is based on edge interrupt count on CSO pin in defined period of time (``ERROR_CHECK_TIMEOUT_MS``) and CSO pin state while work is triggered.
```

Comment:
```suggestion
The check is based on the edge interrupt count on the CSO pin in a defined period of time (``ERROR_CHECK_TIMEOUT_MS``) and on the CSO pin state while the work structure is triggered.
```

Reply:
I think it should be more like work is triggered (not work structure) - I will change.